RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc

```diff
@@ -149,6 +155,26 @@ void PrimaryVertexProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
     edm::LogError("UnusableBeamSpot") << "Beamspot with invalid errors " << beamVertexState.error().matrix();
   }
 
+  //if this is a recovery iteration, check if we already have a valid PV
+  if (isRecoveryIteration) {
+    edm::Handle<reco::VertexCollection> oldPVs;
+    iEvent.getByToken(recoveryVtxToken, oldPVs);
+    const reco::VertexCollection* oldVertices;
+    oldVertices = oldPVs.product();
```

Comment:
```suggestion
    auto const& oldVertices = iEvent.get(recoveryVtxToken);
```
more modern less verbose syntax